### ISSUE TITLE
docker-free installation process?

### ISSUE THREAD

**Issue by bish0polis** [hooram/ownphotos-backend#141]
Hey everyone,

I suspect everyone's busy, but if anyone's installed this project without going the docker route, I'd love to see the docs for that.

No rush, just logging this for later.

**Comment by Derkades** [comment]
I'm not sure why you'd want to do this. Anyway, you should be able to read the [Dockerfile](https://github.com/hooram/ownphotos/blob/dev/Dockerfile) as if it were installation instructions

**Comment by bish0polis** [comment]
> I'm not sure why you'd want to do this

I smell an opportunity to learn why docker is the super-shiny Open Plan Office of the services space.

Good idea, though, even if it'll put a delay on things.  I usually un-bake a docker image through rendering for vagrant and then to a tiny VM - Alpine is fantastic here for its better booting - and this should be no different.  Just, without an established paint-by-numbers, it'll have to go into the 'whenever I can' hopper with the rest of it.  

Maybe keep this ticket open until me or someone else does that bit.  Thanks, though.

**Comment by pepelevamp** [comment]
i want to look at using this tool but the fact that it requires docker and half of a universe to actually run is really off-putting. it just seems like its not a program, but half of a guys computer - just for one program to work.

it would be great if this was made more minimal & self-contained and a bit more sane & didnt require all this crazy environment stuff to run.

it shouldnt really need to run in a docker. its not really a way to make tidy software - its a way to make messy software work.

**Comment by Derkades** [comment]
> it would be great if this was made more minimal & **self-contained** and a bit more sane & didnt require all this crazy environment stuff to run.

That's exactly the point of a docker *container*! Keep in mind that docker containers are not like VMs. There is no overhead to running something in a docker container.

By using a container to run this application you don't need to worry about all its services, dependencies, whatever. They ship you a fully set up installation you can run with a single command!

> its not really a way to make tidy software - its a way to make messy software work.

This is a good point. However, sometimes software is "messy". Sometimes software can't (or shouldn't) be compiled to a single binary. I wouldn't say ownphotos is too crazy, except for maybe the "proxy" server that seems unnecessary. It's a database server (postgres), caching server (redis), backend, frontend and proxy.

Download the docker-compose.yml file, configure it, then run `docker-compose up -d` and the application is running within a couple of minutes. Save yourself the hassle of installing everything manually.

**Comment by bish0polis** [comment]
Yep, it's shiny, no doubt about that.  And it's surely better than installing everything 'manually', which isn't done for 99% of software past 1995.  The merits of Docker vs last week's darling is something largely discussed elsewhere, and if Docker isn't the product then most of that debate isn't relevant here but for one issue at-hand:

Docker's not allowed in some organizations for safety, security, reporting and validation problems.  Also, for some people it's not their thing because some wart the aficionados see as minor is a really big deal for others.

I'm in the category of people who don't see a path to where docker's ever in use, but would like to use this software.  Please keep this ticket open in the hopes that a docker-free installation will be part of the maintained documentation.

**Comment by pepelevamp** [comment]
the problem becomes - when you want to install this properly, you are left with two choices: 
either install it fully and deal with how messy it is.
or
install the whole universe that comes with it including whatever OS shim like ubuntu 16 etc.

when the software is distributed on its own, and as a docker image, then yeah ya get a try before ya buy thing thats wonderful. 
but docker isn't a catch-all solution thats appropriate for every installation type. it shouldn't be a way to deliver/ship software. you can't have 500 OSes for 500 programs.

 i dont want some wacky OS running on my system just so i can have a program work. "it cant affect the rest of your system" is true, but so what? its still a huge waste and basically someone else's OS on your hardware & burning CPU cycles

**Comment by DigitalWarrior** [comment]
The Development of Ownphotos has stopped. LibrePhotos is a fork that is actively being developed. You can find it here https://github.com/LibrePhotos/librephotos. For a non docker install follow the instructions here https://github.com/LibrePhotos/librephotos-linux

**Comment by bish0polis** [comment]
TLDR?  PIP and Debs.  Two validation hits.  But thanks for the excellent pointer -- we can hopefully introduce validation and checksums and Single Source of Truth for software state, since the project is still malleable.  Excellent !!
